dask_expr/_collection.py

```diff
@@ -1810,6 +1816,58 @@ def from_graph(*args, **kwargs):
     return new_collection(FromGraph(*args, **kwargs))
 
 
+def from_dict(
+    data,
+    npartitions,
+    orient="columns",
+    dtype=None,
+    columns=None,
+    constructor=pd.DataFrame,
+):
+    """
+    Construct a Dask DataFrame from a Python Dictionary
+
+    Parameters
+    ----------
+    data : dict
+        Of the form {field : array-like} or {field : dict}.
+    npartitions : int
+        The number of partitions of the index to create. Note that depending on
+        the size and index of the dataframe, the output may have fewer
+        partitions than requested.
+    orient : {'columns', 'index', 'tight'}, default 'columns'
+        The "orientation" of the data. If the keys of the passed dict
+        should be the columns of the resulting DataFrame, pass 'columns'
+        (default). Otherwise if the keys should be rows, pass 'index'.
+        If 'tight', assume a dict with keys
+        ['index', 'columns', 'data', 'index_names', 'column_names'].
+    dtype: bool
+        Data type to force, otherwise infer.
+    columns: string, optional
+        Column labels to use when ``orient='index'``. Raises a ValueError
+        if used with ``orient='columns'`` or ``orient='tight'``.
+    constructor: class, default pd.DataFrame
+        Class with which ``from_dict`` should be called with.
+
+    Examples
+    --------
+    >>> import dask.dataframe as dd
+    >>> ddf = dd.from_dict({"num1": [1, 2, 3, 4], "num2": [7, 8, 9, 10]}, npartitions=2)
+    """
+
+    collection_types = {type(v) for v in data.values() if is_dask_collection(v)}
```

Comment:
nit: The error message might be more helpful if we stored these as a dict to highlight where the collections were provided.

Reply:
I agree generally speaking, but this can produce very long error messages that screw your terminal (we had cases like this in pandas)